qrserve.go

```diff
@@ -71,7 +72,11 @@ func qrHandler(w http.ResponseWriter, req *http.Request) {
 
 	// Serve the resulting PNG image to the client and we're done!
 	w.Header().Set("Content-Type", "image/png")
-	w.Write(image)
+	_, err = w.Write(image)
+	if err != nil {
+		http.Error(w, "Error writing image: "+err.Error(), 500)
```

Comment:
I can’t recall any case where `w.Write()` would error out but `http.Error` would work?, esp. since if Write partially succeeded, headers could no longer be modified. In that case it may make more sense to log the error to the console.